### PR TITLE
TIG-1359 use genny toolchain for compilation

### PR DESCRIPTION
**Pull request by guoyr**
Waiting for the patch build to finish first before requesting a review

### CODE REVIEW

File: scripts/bootstrap.sh [hunk 1]
```diff
@@ -13,9 +13,6 @@ BUILD_DIR="${ROOT_DIR}/build"
 cd "${BUILD_DIR}"
 rm -rf ./*
 
-# shellcheck disable=SC1090
-. "${SCRIPTS_DIR}/find-openssl.rc"
```
Comment (bcaimano) [inline scripts/bootstrap.sh]: I think this may or may not matter to @spencerjackson with his Atlas work right now.

File: evergreen.yml [hunk 2]
```diff
@@ -359,23 +347,24 @@ functions:
...
       working_dir: src
       binary: /bin/bash
       args:
-      - ${workdir}/src/scripts/bootstrap.sh
+      - "${workdir}/src/scripts/bootstrap.sh"
+      - "-DCMAKE_PREFIX_PATH=${workdir}/toolchain/shared/x64-${os_family}-shared"
+      - "-DCMAKE_TOOLCHAIN_FILE=${workdir}/toolchain/static/scripts/buildsystems/vcpkg.cmake"
+      - "-DVCPKG_TARGET_TRIPLET=x64-${os_family}-static"
       - "-DENABLE_JASPER:BOOL=ON"
-      - "-DGENNY_STATIC_BOOST_PATH:PATH=${workdir}/boost"
```
Comment (bcaimano) [inline evergreen.yml]: Are we actually getting static boost or are we now defaulting to shared?

File: evergreen.yml [hunk 3]
```diff
@@ -359,23 +347,24 @@ functions:
...
       - "-DGENNY_INSTALL_DIR:PATH=${workdir}/dist"
       - "-DGENNY_INSTALL_SSL:BOOL=ON"
       env:
         # This should be reasonably safe even on low memory machines
         MAKEFLAGS: "-j8"
         CMAKE: ${cmake|cmake}
-        PATH: ${toolchain}:/usr/bin:/bin
```
Comment (bcaimano) [inline evergreen.yml]: ...I forget why this PATH was necessary.

File: scripts/bootstrap.sh (hunk 1 above)
Comment (guoyr) [reply scripts/bootstrap.sh]: Good to know! You can still set `OPENSSL_ROOT_DIR` to override the toolchain openssl if that's the use case you had in mind.

File: evergreen.yml (hunk 2 above)
Comment (guoyr) [reply evergreen.yml]: I'm pretty sure we're using static boost:

```
[root@ip-10-122-9-172 build]# ldd src/driver/genny 
	linux-vdso.so.1 =>  (0x00007ffe848d5000)
	libcast_core.so => /data/mci/genny/build/src/cast_core/libcast_core.so (0x00007f80c63fd000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f80c61d1000)
	libmongocxx.so._noabi => /data/mci/genny/shared/installed/x64-linux-shared/lib/libmongocxx.so._noabi (0x00007f80c5f2c000)
	libbsoncxx.so._noabi => /data/mci/genny/shared/installed/x64-linux-shared/lib/libbsoncxx.so._noabi (0x00007f80c5bd5000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f80c58d2000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f80c56bc000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f80c52f9000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f80c6b82000)
	libmongoc-1.0.so.0 => /data/mci/genny/shared/installed/x64-linux-shared/lib/libmongoc-1.0.so.0 (0x00007f80c4e12000)
	libbson-1.0.so.0 => /data/mci/genny/shared/installed/x64-linux-shared/lib/libbson-1.0.so.0 (0x00007f80c4bd7000)
	libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007f80c49b9000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f80c47b1000)
	libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f80c4597000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f80c4380000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f80c417c000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f80c3f45000)
	libfreebl3.so => /lib64/libfreebl3.so (0x00007f80c3d41000)

[root@ip-10-122-9-172 build]# ldd src/cast_core/libcast_core.so 
	linux-vdso.so.1 =>  (0x00007fff96d88000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f6a321b0000)
	libmongocxx.so._noabi => /data/mci/genny/shared/installed/x64-linux-shared/lib/libmongocxx.so._noabi (0x00007f6a31f0a000)
	libbsoncxx.so._noabi => /data/mci/genny/shared/installed/x64-linux-shared/lib/libbsoncxx.so._noabi (0x00007f6a31bb3000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f6a318b1000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f6a3169a000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f6a312d7000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6a32b60000)
	libmongoc-1.0.so.0 => /data/mci/genny/shared/installed/x64-linux-shared/lib/libmongoc-1.0.so.0 (0x00007f6a30df0000)
	libbson-1.0.so.0 => /data/mci/genny/shared/installed/x64-linux-shared/lib/libbson-1.0.so.0 (0x00007f6a30bb5000)
	libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007f6a30998000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f6a30790000)
	libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f6a30575000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f6a3035f000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f6a3015a000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f6a2ff23000)
	libfreebl3.so => /lib64/libfreebl3.so (0x00007f6a2fd20000)
```

File: evergreen.yml (hunk 3 above)
Comment (guoyr) [reply evergreen.yml]: Arh, because we need the compiler from the mongodb toolchain. I'm going to put it back in

File: evergreen.yml (hunk 2 above)
Comment (bcaimano) [reply evergreen.yml]: I believe it would be libgennylib.so that would have the links.

Comment (guoyr) [reply evergreen.yml]: `libgennylib.a` is (has been?) statically linked.

I also double-checked that boost** symbols are indeed defined:
```
➜  build git:(tig1359) ✗ nm src/gennylib/libgennylib.a | grep " T .*boost" | wc
     371    1113   45622
```

**Comment by guoyr** [comment]
Closing this in favor of #158 with a single command genny compile script

**Pull request closed without merging**